src/utils/compare.ts

```diff
@@ -0,0 +1,20 @@
+import { hexToRgb } from "./convert";
+
+export const equalColorObjects = (
+  first: Record<string, unknown>,
+  second: Record<string, unknown>
+): boolean => {
+  if (first === second) return true;
+
+  for (const prop in first) {
+    if (first[prop] !== second[prop]) return false;
+  }
+
+  return true;
+};
+
+export const equalHex = (first: string, second: string): boolean => {
+  if (first.toLowerCase() === second.toLowerCase()) return true;
+
+  return equalColorObjects(hexToRgb(first), hexToRgb(second));
```

Comment:
I'm confused as towards why this is here. Surely if the hex strings are not equal converting them to RGB isn't going to change this fact?

Can this just be removed? Or is there an edge case I'm not thinking of?

Reply:
react-colorful support HEX3 colors, so it's possible that user will pass a color like `#ABC` and `"#abc".toLowerCase() !== "#aabbcc".toLowerCase())` so I just convert both colors to RGB to compare they RGB representations. This way is smallest in terms of bundle size baucause `hexToRgb` is in the bundle anyway.

Reply:
Actually HEX is the same as RGB but formatted differently.
Both models include thee [0-255] numbers. RGB store them in decimal, HEX — hexadecimal (00-ff where ff = 255).

Reply:
Oh, true. Sorry. Forgot about HEX3/HEX6. Makes sense.

Though I wonder if it would be better to avoid conversion to RGB and the iterating over object indexes altogether.

Reply:
No problem)
BTW this code also solves the case when one of the colors doesn't have `#`. Like `equalHex('000', '#000)`. 
`hexToRgb` knows everything about hex-strings anatomy =)

Reply:
Yeah fair. I'll leave it. If there's performance improvements to be had this PR is not the place to do it. Already too much going on.

Reply:
Agree. PR is huge 😀
It's amazing how much you have done.

Reply:
Sorry about that. Hard to incrementally move to TS.

Reply:
Don't worry! Actually you're a hero) And I don't know how to thank you) Feel free to invite me to any of your projects — I'll be glad to help. I'm not good in terms TS yet, but I'm in web-dev since 2008 and know a lot of stuff =)